server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGateFinder.java

```diff
@@ -46,16 +47,16 @@ public QualityGateFinder(DbClient dbClient) {
    * It will first try to get the quality gate explicitly defined on a project, if none it will try to return default quality gate.
    * As it's possible to have no default quality gate, this method can return {@link Optional#empty()}
    */
-  public Optional<QualityGateData> getQualityGate(DbSession dbSession, long componentId) {
+  public Optional<QualityGateData> getQualityGate(DbSession dbSession, OrganizationDto organization, long componentId) {
     Optional<Long> qualityGateId = dbClient.projectQgateAssociationDao().selectQGateIdByComponentId(dbSession, componentId);
     if (qualityGateId.isPresent()) {
       return Optional.of(new QualityGateData(getById(dbSession, qualityGateId.get()), false));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "getById"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)